modules/navier_stokes/src/userobjects/INSFVRhieChowInterpolator.C

```diff
@@ -601,6 +645,170 @@ INSFVRhieChowInterpolator::getVelocity(const Moose::FV::InterpMethod m,
+    Real interp_vf;
+    Real uncorrected_interp_vf;
+
+    // Compute the corrected interpolated face value
+    if (Moose::FV::onBoundary(*this, fi))
+    {
+      const Elem * const boundary_elem = hasBlocks(elem->subdomain_id()) ? elem : neighbor;
+      const Moose::FaceArg boundary_face{
+          &fi, Moose::FV::LimiterType::CentralDifference, true, correct_skewness, boundary_elem};
+
+      interp_vf = 0.0;
+      for (const auto i : make_range(_volumetric_force.size()))
+        interp_vf += (*this->_volumetric_force[i])(boundary_face, time);
+    }
+    else
+    {
+      Moose::FaceArg face{
+          &fi, Moose::FV::LimiterType::CentralDifference, true, correct_skewness, nullptr};
+
+      interp_vf = 0.0;
+      for (const auto i : make_range(_volumetric_force.size()))
+        interp_vf += (*this->_volumetric_force[i])(face, time);
+    }
+
+    // Compute the uncorrected interpolated face value
+    // For it to be consistent with the pressure gradient intepolation `uncorrectedAdGradSln`
+    // it should follow the same process than Green Gauss but for a volume force
+
+    Real elem_value = 0.0;
+    Real neigh_value = 0.0;
+
+    // Uncorrected interpolation - Step 1: loop over the faces of the element to compute
+    // face-average cell value
+    for (const auto side : make_range(elem->n_sides()))
+    {
+      const Elem * const loc_neighbor = elem->neighbor_ptr(side);
+      const bool elem_has_fi = Moose::FV::elemHasFaceInfo(*elem, loc_neighbor);
+      const FaceInfo * const fi_loc =
+          _moose_mesh.faceInfo(elem_has_fi ? elem : loc_neighbor,
+                               elem_has_fi ? side : loc_neighbor->which_neighbor_am_i(elem));
+
+      if (Moose::FV::onBoundary(*this, *fi_loc))
+      {
+        const Elem * const boundary_elem = hasBlocks(elem->subdomain_id()) ? elem : loc_neighbor;
+        const Moose::FaceArg loc_boundary_face{fi_loc,
+                                               Moose::FV::LimiterType::CentralDifference,
+                                               true,
+                                               correct_skewness,
+                                               boundary_elem};
+
+        for (const auto i : make_range(_volumetric_force.size()))
+          elem_value += (*this->_volumetric_force[i])(loc_boundary_face, time) *
+                        fi_loc->faceArea() *
+                        (fi_loc->faceCentroid() - elem->vertex_average()).norm() *
+                        (fi_loc->normal() * loc_normal);
+      }
+      else
+      {
+        Moose::FaceArg loc_face{
+            fi_loc, Moose::FV::LimiterType::CentralDifference, true, correct_skewness, nullptr};
+
+        for (const auto i : make_range(_volumetric_force.size()))
+          elem_value += (*this->_volumetric_force[i])(loc_face, time) * fi_loc->faceArea() *
+                        (fi_loc->faceCentroid() - elem->vertex_average()).norm() *
```

Comment:
This is because I am trying to reconstruct the effective volume term from the faces. The volume associated to the face should be faceArea*||face_position - element_position||. Changed the line to `fi_loc->faceCentroid() - elem->centroid()).norm()` to be clear